src/main/resources/db/migration/V17__POL-453-expose-last-triggered-to-policies.sql

```diff
@@ -0,0 +1,13 @@
+ALTER TABLE policy ADD column last_triggered bigint default 0;
+
+UPDATE policy as p SET last_triggered = (SELECT MAX(ctime) FROM policies_history as ph WHERE p.id = uuid(ph.policy_id));
+
+CREATE FUNCTION update_last_triggered() RETURNS TRIGGER AS $update_last_triggered$
+    BEGIN
+        UPDATE policy SET last_triggered = GREATEST(last_triggered, new.ctime) WHERE id = uuid(new.policy_id);
+        RETURN NEW;
+    END;
+$update_last_triggered$ LANGUAGE plpgsql;
+
+CREATE TRIGGER policies_update_last_triggered AFTER INSERT ON policies_history
+    FOR EACH ROW EXECUTE PROCEDURE update_last_triggered();
```

Comment:
This could probably be a bit simpler with a [Postgres rule](https://www.postgresql.org/docs/current/rules-update.html).

Untested:
```suggestion
CREATE RULE rule_policies_history AS ON INSERT TO policies_history
DO UPDATE policy
SET last_triggered = GREATEST(last_triggered, NEW.ctime)
WHERE id = uuid(NEW.policy_id);
```

Reply:
Is there any reason to use this syntax (other than it looks simpler?)
At first glance on the docs it says: 
```
Caution
In many cases, tasks that could be performed by rules on INSERT/UPDATE/DELETE are better done with triggers.
```

I'll take a look as it looks interesting anyway.

Reply:
Rules can be slower for complex queries when the query planner job is critical for performances. But this one is not complex so I don't think this applies here.

Reply:
[This](https://www.postgresql.org/docs/current/rules-triggers.html) is another interesting article to read about rules and triggers.